src/main/java/winstone/Launcher.java

```diff
@@ -229,9 +231,19 @@ private void writePortToFileIfNeeded() throws IOException {
                     Path portFile = Paths.get(portFileName);
                     Path portDir = portFile.getParent();
                     Files.createDirectories(portDir);
-                    try (BufferedWriter writer = Files.newBufferedWriter(portFile, StandardCharsets.UTF_8)) {
+                    Path tmpPath = Files.createTempFile(portDir, "temp", "");
+                    try (BufferedWriter writer = Files.newBufferedWriter(tmpPath, StandardCharsets.UTF_8)) {
                         writer.write(Integer.toString(port));
                     }
+
+                    try {
+                        Files.move(tmpPath, portFile, StandardCopyOption.ATOMIC_MOVE);
+                    } catch (AtomicMoveNotSupportedException e) {
+                        Logger.logDirectMessage(Level.WARNING, "", "Atomic move does not supported", e);
```

Comment:
Elsewhere in this file, the `streamName` argument is always passed as `null` and `Logger.WARNING` is used rather than `Level.WARNING`, so for consistency I think we should do the same here. Also, I think the message could be improved to state that we are about to make another attempt:

```suggestion
                        Logger.logDirectMessage(Logger.WARNING, null, "Atomic move not supported. Falling back to non-atomic move.", e);
```